docs/recipes/integration-test/example-lh-auth.test.js

```diff
@@ -0,0 +1,183 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/**
+ * @fileoverview Example Jest tests for demonstrating how to run Lighthouse on an authenticated
+ * page as integration tests. See docs/recipes/auth/README.md for more.
+ */
+
+/** @typedef {import('./node_modules/lighthouse/types/lhr')} LH */
+
+const puppeteer = require('puppeteer');
+const lighthouse = require('lighthouse');
+const server = require('../auth/server/server.js');
+
+const CHROME_DEBUG_PORT = 8042;
+const SERVER_PORT = 8000;
+
+jest.setTimeout(30000);
+
+// Provide a nice way to assert a score for a category.
+// Note, you could just use `expect(lhr.categories.seo.score).toBeGreaterThanOrEqual(0.9)`,
+// but by using a custom matcher a better error report can be generated.
+expect.extend({
+  toHaveLighthouseScoreGreaterThanOrEqual(lhr, category, threshold) {
+    const score = lhr.categories[category].score;
+    const auditsRefsByWeight = [...lhr.categories[category].auditRefs]
+      .filter((auditRef) => auditRef.weight > 0)
+      .sort((a, b) => b.weight - a.weight);
+    const report = auditsRefsByWeight.map((auditRef) => {
+      const audit = lhr.audits[auditRef.id];
+      const status = audit.score === 1 ?
+        this.utils.EXPECTED_COLOR('○') :
+        this.utils.RECEIVED_COLOR('✕');
+      const weight = this.utils.DIM_COLOR(`[weight: ${auditRef.weight}]`);
+      return `\t${status} ${weight} ${audit.id}`;
+    }).join('\n');
+
+    if (score >= threshold) {
+      return {
+        pass: true,
+        message: () =>
+          `expected category ${category} to be < ${threshold}, but got ${score}\n${report}`,
+      };
+    } else {
+      return {
+        pass: false,
+        message: () =>
+          `expected category ${category} to be >= ${threshold}, but got ${score}\n${report}`,
+      };
+    }
+  },
+});
+
+/**
+ * @param {string} url
+ * @return {Promise<LH.Result>}
+ */
+async function runLighthouse(url) {
+  const result = await lighthouse(url, {
+    port: CHROME_DEBUG_PORT,
+    onlyCategories: ['seo'],
+  });
+  return result.lhr;
+}
+
+/**
+ * @param {puppeteer.Browser} browser
+ */
+async function login(browser) {
```

Comment:
shouldnt this function come from example-lh-auth.js ?

Reply:
ya, moved